src/Compilers/CSharp/Test/Emit3/FirstClassSpanTests.cs

```diff
@@ -0,0 +1,1931 @@
+              IL_0005:  ldloc.0
+              IL_0006:  newobj     "System.Span<string>..ctor(string[])"
+              IL_000b:  ret
+            }
+            """);
+        verifier.VerifyIL("C.M2", """
+            {
+              // Code size       12 (0xc)
+              .maxstack  3
+              .locals init (string[] V_0)
+              IL_0000:  ldarg.1
+              IL_0001:  ldnull
+              IL_0002:  dup
+              IL_0003:  stloc.0
+              IL_0004:  stind.ref
+              IL_0005:  ldloc.0
+              IL_0006:  newobj     "System.ReadOnlySpan<string>..ctor(string[])"
+              IL_000b:  ret
+            }
+            """);
+    }
+
+    [Fact]
+    public void Conversion_Array_Span_Implicit_Value()
+    {
+        var source = """
+            class C
+            {
+                void M()
+                {
+                    System.Span<string> s = A();
+                    System.ReadOnlySpan<string> r = A();
+                }
+                string[] A() => null;
+            }
+            """;
+        var comp = CreateCompilationWithSpan(source, TestOptions.DebugDll);
+        var verifier = CompileAndVerify(comp).VerifyDiagnostics();
+        verifier.VerifyIL("C.M", """
+            {
+              // Code size       28 (0x1c)
+              .maxstack  2
+              .locals init (System.Span<string> V_0, //s
+                            System.ReadOnlySpan<string> V_1) //r
+              IL_0000:  nop
+              IL_0001:  ldloca.s   V_0
+              IL_0003:  ldarg.0
+              IL_0004:  call       "string[] C.A()"
+              IL_0009:  call       "System.Span<string>..ctor(string[])"
+              IL_000e:  ldloca.s   V_1
+              IL_0010:  ldarg.0
+              IL_0011:  call       "string[] C.A()"
+              IL_0016:  call       "System.ReadOnlySpan<string>..ctor(string[])"
+              IL_001b:  ret
+            }
+            """);
+    }
+
+    [Theory, MemberData(nameof(LangVersions))]
+    public void Conversion_Array_Span_Opposite_Implicit(LanguageVersion langVersion)
+    {
+        var source = """
+            class C
+            {
+                 int[] M(System.Span<int> arg) => arg;
+            }
+            """;
+        CreateCompilationWithSpan(source, parseOptions: TestOptions.Regular.WithLanguageVersion(langVersion)).VerifyDiagnostics(
+            // (3,39): error CS0029: Cannot implicitly convert type 'System.Span<int>' to 'int[]'
+            //      int[] M(System.Span<int> arg) => arg;
+            Diagnostic(ErrorCode.ERR_NoImplicitConv, "arg").WithArguments("System.Span<int>", "int[]").WithLocation(3, 39));
+    }
+
+    [Theory, MemberData(nameof(LangVersions))]
+    public void Conversion_Array_Span_Opposite_Explicit(LanguageVersion langVersion)
+    {
+        var source = """
+            class C
+            {
+                 int[] M(System.Span<int> arg) => (int[])arg;
+            }
```

Comment:
>} [](http://example.com/codeflow?start=12&length=1)

Consider testing `int[]` to `ReadOnlySpan<int>` as well, perhaps in the same test.